src/main/java/weavers/siltarae/login/domain/GoogleProvider.java

```diff
@@ -0,0 +1,74 @@
+package weavers.siltarae.login.domain;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import weavers.siltarae.global.exception.AuthException;
+import weavers.siltarae.global.exception.ExceptionCode;
+import weavers.siltarae.login.dto.request.OAuthAccessTokenRequest;
+import weavers.siltarae.login.dto.response.OAuthAccessTokenResponse;
+import weavers.siltarae.login.dto.response.MemberInfoResponse;
+import weavers.siltarae.member.domain.SocialType;
+
+import java.net.URI;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class GoogleProvider extends OAuthProvider {
+
+    private final GoogleAuthClient googleAuthClient;
+
+    private final String SOCIAL_TYPE = "GOOGLE";
+
+    @Value("${oauth2.google.client-id}")
+    private String CLIENT_ID;
+
+    @Value("${oauth2.google.client-secret}")
+    private String CLIENT_SECRET;
+
+    @Value("${oauth2.google.redirect-uri}")
+    private String REDIRECT_URI;
+
+    @Value("${oauth2.google.grant-type}")
+    private String GRANT_TYPE;
+
+    @Value("${oauth2.google.member-info-base-uri}")
+    private String MEMBER_INFO_BASE_URI;
+
+    @Override
+    public MemberInfoResponse getMemberInfo(String accessToken) {
+
+        ResponseEntity<MemberInfoResponse> response = googleAuthClient.getMemberInfo(URI.create(MEMBER_INFO_BASE_URI), accessToken);
+
+        if (!response.getStatusCode().is2xxSuccessful()) {
+            throw new AuthException(ExceptionCode.NOT_SUPPORTED_AUTH_SERVICE);
+        }
+
+        MemberInfoResponse memberInfo = response.getBody();
+        memberInfo.setSocialType(SocialType.GOOGLE);
+
+        return memberInfo;
+    }
+
+    @Override
+    public String requestAccessToken(final String code) {
+
+        OAuthAccessTokenRequest request = OAuthAccessTokenRequest.builder()
+                .clientId(CLIENT_ID)
+                .clientSecret(CLIENT_SECRET)
+                .redirectUri(REDIRECT_URI)
+                .grantType(GRANT_TYPE)
+                .code(code).build();
+
+        ResponseEntity<OAuthAccessTokenResponse> response = googleAuthClient.getAccessToken(request);
+
+        if(!response.getStatusCode().is2xxSuccessful()) {
+            throw new AuthException(ExceptionCode.INVALID_AUTHORIZATION_CODE);
+        }
+
+        return response.getBody().getAccessToken();
```

Comment:
!response.getStatusCode().is2xxSuccessful() 이부분에서 성공 여부를 판별해주니깐 response.getBody()가 null 일 가능성은 없는 것 인가요?

Reply:
그 부분은 구글 서버 측에서 응답을 던져주는 거긴 한데... 검색해보니 구글 API에서 null을 반환한다는 질문글이 있긴 하네요.
위 로직이랑 비슷하게 바꿔볼까요?

```suggestion
        if(!response.getStatusCode().is2xxSuccessful() || !response.hasBody()) {
            throw new AuthException(ExceptionCode.INVALID_AUTHORIZATION_CODE);
        }

        return response.getBody().getAccessToken();
```

Reply:
좋아요~~ 👍